servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultHost.java

```diff
@@ -91,25 +92,36 @@ private enum State {
     private final HealthIndicator<Addr, C> healthIndicator;
     private final LoadBalancerObserver.HostObserver hostObserver;
     private final ConnectionFactory<Addr, ? extends C> connectionFactory;
+    private final int minConnections;
     private final ListenableAsyncCloseable closeable;
     private volatile ConnState connState = new ConnState(emptyList(), State.ACTIVE, 0, null);
 
     DefaultHost(final String lbDescription, final Addr address,
                 final ConnectionSelector<C> connectionSelector,
                 final ConnectionFactory<Addr, ? extends C> connectionFactory,
+                final int minConnections,
                 final HostObserver hostObserver, @Nullable final HealthCheckConfig healthCheckConfig,
                 @Nullable final HealthIndicator<Addr, C> healthIndicator) {
         this.lbDescription = requireNonNull(lbDescription, "lbDescription");
         this.address = requireNonNull(address, "address");
         this.healthIndicator = healthIndicator;
         this.connectionSelector = requireNonNull(connectionSelector, "connectionSelector");
         requireNonNull(connectionFactory, "connectionFactory");
+        this.minConnections = ensureNonNegative(minConnections, "minConnections");
         this.connectionFactory = healthIndicator == null ? connectionFactory :
                 new InstrumentedConnectionFactory<>(connectionFactory, healthIndicator);
         assert healthCheckConfig == null || healthCheckConfig.failedThreshold > 0;
         this.healthCheckConfig = healthCheckConfig;
         this.hostObserver = requireNonNull(hostObserver, "hostObserver");
         this.closeable = toAsyncCloseable(this::doClose);
+
+        // warm the connection pool.
+        // TODO: this will likely be racy with initial requests also wanting to create connections.
+        //   At a minimum, we should perhaps warm only a single connection at a time.
+        for (int i = 0; i < minConnections; i++) {
+            newConnection(cxn -> true, false, null)
+                    .toFuture();
+        }
```

Comment:
This will get a bit more work to consolidate and hopefully make it a bit cleaner.